modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -1665,18 +1733,39 @@ public CommerceOrder updateCommerceOrder(
 			commerceOrderId);
 
 		commerceOrder.setExternalReferenceCode(externalReferenceCode);
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannelByGroupId(
+				commerceOrder.getGroupId());
+
 		commerceOrder.setBillingAddressId(billingAddressId);
-		commerceOrder.setCommerceShippingMethodId(commerceShippingMethodId);
+
+		_setAccountDefaultBillingAddress(commerceChannel, commerceOrder);
+
 		commerceOrder.setShippingAddressId(shippingAddressId);
+
+		_setAccountDefaultShippingAddress(commerceChannel, commerceOrder);
+
+		User user = _userLocalService.getUser(commerceOrder.getUserId());
+
+		commerceOrder.setCommerceShippingMethodId(commerceShippingMethodId);
+		commerceOrder.setShippingOptionName(shippingOptionName);
+
+		_setAccountDefaultShippingOption(commerceChannel, commerceOrder, user);
+
```

Comment:
```

		commerceOrder.setCommerceShippingMethodId(commerceShippingMethodId);
		commerceOrder.setShippingOptionName(shippingOptionName);

		_setAccountDefaultShippingOption(commerceChannel, commerceOrder, user);
```

Was intentionally placed after setter of the `billingAddress` and `shippingAddress` because we depend on the two values for the rest of the logic.